docs/manual/howto/widget.rst

```diff
@@ -0,0 +1,496 @@
+The key difference is that the ``poll`` method is run asynchronously and triggers a
+callback once the function completes. This allows widgets to get text from
+long-running functions without blocking Qtile.
+
+Mixins
+======
+
+As well as inheriting from one of the base classes above, widgets can also inherit one
+or more mixins to provide some additional functionality to the widget.
+
+PaddingMixin
+------------
+
+This provides the ``padding(_x|_y|)`` attributes which can be used to change the appearance
+of the widget.
+
+If you use this mixin in your widget, you need to add the following line to your ``__init__``
+method:
+
+.. code:: python
+
+    self.add_defaults(base.PaddingMixin.defaults)
+
+MarginMixin
+-----------
+
+The ``MarginMixin`` is essentially effectively exactly the same as the ``PaddingMixin`` but,
+instead, it provides the ``margin(_x|_y|)`` attributes.
+
+As above, if you use this mixin in your widget, you need to add the following line to your
+``__init__`` method:
+
+.. code:: python
+
+    self.add_defaults(base.MarginMixin.defaults)
+
+Configuration
+=============
+
+Now you know which class to base your widget on, you need to know how the widget
+gets configured.
+
+Defining Parameters
+-------------------
+
+Each widget will likely have a number of parameters that users can change to
+customise the look and feel and/or behaviour of the widget for their own needs.
+
+The widget should therefore provide the default values of these parameters as a
+class attribute called ``defaults``. The format of this attribute is a list of
+tuples.
+
+.. code:: python
+
+    defaults = [
+        ("parameter_name",
+         default_parameter_value,
+         "Short text explaining what parameter does")
+    ]
+
+Users can override the default value when creating their ``config.py`` file.
+
+.. code:: python
+
+    MyCustomWidget(parameter_name=updated_value)
+
+Once the widget is initialised, these parameters are available at
+``self.parameter_name``.
+
+The __init__ method
+-------------------
+
+Parameters that should not be changed by users can be defined in the ``__init__``
+method.
+
+This method is run when the widgets are initially created. This happens before
+the ``qtile`` object is available. This section is therefore appropriate for
+hard coding values.
+
+The _configure method
+---------------------
+
+The ``_configure`` method is called by the ``bar`` object and sets the
+``self.bar`` and ``self.qtile`` attributes of the widget. It also creates the
+``self.drawer`` attribute which is necessary for displaying any content.
+
+Once this method has been run, your widget should be ready to display content as
+the bar will draw once it has finished its configuration.
+
+Calls to methods required to prepare the content for your widget should therefore
+be made from this method rather than ``__init__``.
+
+Displaying output
+=================
+
+As noted at the beginning of this page, a Qtile widget is just a drawing that is
```

Comment:
I think you can drop the first bits before the comma.

Reply:
Agreed.